src/components/SeatMap/SeatMap.test.js

```diff
@@ -0,0 +1,554 @@
+import { render, screen, waitFor, act } from '@testing-library/react';
+import { JetsSeatMap } from './SeatMap';
+import { CONFIG_MOCK } from '../Demo/constants';
+import { flightDetails } from './__fixtures__/seatMapApiGetPlaneFeatures';
+import { finalDeck, finalSeat, passenger } from './__fixtures__/SeatMapService';
+import { paramsData } from '../TooltipGlobal/__fixtures__';
+import { bulk, exit, row, seat } from './__fixtures__/seatMapApiPostDataResponse';
+import { JetsSeatMapService } from './service';
+
+jest.mock('./service');
+
+const setup = ({
+  flight,
+  availability,
+  passengers,
+  currentDeckIndex,
+  configOverrides,
+  onSeatMapInited,
+  onAvailabilityApplied,
+}) => {
+  const config = {
+    ...CONFIG_MOCK,
+    ...configOverrides,
+  };
+  return render(
+    <JetsSeatMap
+      flight={flight}
+      availability={availability}
+      passengers={passengers}
+      currentDeckIndex={currentDeckIndex}
+      config={config}
+      onSeatMapInited={onSeatMapInited}
+      onAvailabilityApplied={onAvailabilityApplied}
+    />
+  );
+};
+
+describe('JetsSeatMap', () => {
+  beforeEach(() => {
+    JetsSeatMapService.prototype = {
+      getSeatMapData: jest.fn(),
+      addAbbrToPassengers: jest.fn(),
+      setPassengersHandler: jest.fn(),
+      setAvailabilityHandler: jest.fn(),
+      compareWithDecksSeatsInfo: jest.fn(),
+    };
+  });
+
+  describe('when the seat map is rendered', () => {
+    it('should initialize the seat map if all valid data was passed', async () => {
+      const flight = flightDetails();
+      const availability = [{ data: 'some availability' }];
+      const passengers = [{ data: 'some passenger data' }];
```

Comment:
Same as mentioned above, could we use fake data that contains the format that this component will be receiving please? e.g. [{
    "id": "4",
    "seat": null,
    "passengerLabel": “Sam Smith”,
}]